File: ssh/LabyrinthCharacter.rb
#encoding:utf-8

module Irrgarten
	
	class LabyrinthCharacter
	
		protected
			
			def get_health
				return @health
			end
			
			def set_health(h)
				@health = h
			end
			
			def got_wounded
				@health-=1
			end
		
		public
			
			def initialize(n,i,s,h) 
				@name=n
				@intelligence=i
				@strength=s
				@health=h
				@row=nil
				@col=nil
			end
			
			def copy(o)  #PREGUNTAR
				@name = o.get_name
				@intelligence = o.get_intelligence
				@strength = o.get_strength
				@health = o.get_health
				@row = o.get_row
				@col = o.get_col
			end
			
			def dead
				if (@health==0)
					return true
				else
					return false
				end
			end
			
			def get_intelligence
				return @intelligence
			end
			
			def get_strength
				return @strength
			end
			
			def get_name
				return @name
			end
			
			def get_row
				return @row
			end
			
			def get_col
				return @col
			end
			
			def set_pos(r,c)
				@row=r
				@col=c
			end
			
			def to_s
				return "[[#{@name}, #{@intelligence}, #{@strength}, #{@health}, #{@row}, #{@col}]"
			end
	end
end

File: irrgarten/src/irrgarten/LabyrinthCharacter.java
/*
 * Click nbfs://nbhost/SystemFileSystem/Templates/Licenses/license-default.txt to change this license
 * Click nbfs://nbhost/SystemFileSystem/Templates/Classes/Class.java to edit this template
 */
package irrgarten;

/**
 *
 * @author Juan Molina Molina
 * @author Paula Jiménez Ruiz
 */
abstract class LabyrinthCharacter {
    
    private String name;
    private float intelligence;
    private float strenght;
    private float health;
    private int row;
    private int col;
    
    public LabyrinthCharacter(String name, float intelligence, float strenght, float health) {
    
        this.name=name;
        this.intelligence=intelligence;
        this.strenght=strenght;
        this.health=health;
    }
    
    public LabyrinthCharacter(LabyrinthCharacter other) {  // Sintaxis importante
    
        this(other.name, other.intelligence, other.strenght, other.health);
        this.row = other.row;
        this.col = other.col;
    }
    
    public boolean dead() {
    
        if(health==0) {
            return true;
        }
        else
            return false;
    }
    
    public int getRow() {
    
        return row;
    }
    
    public int getCol() {
    
        return col;
    }
    
    protected float getIntelligence() {
    
        return intelligence;
    }
    
    protected float getStrength() {
    
        return strenght;
    }
    
    protected float getHealth() {
    
        return health;
    }
    
    protected void setHealth(float health) {
    
        this.health=health;
    }
    
    public void setPos(int row, int col) {
    
        this.row=row;
        this.col=col;
    }
    
    @Override
    public String toString() {
    
        return "["+name+", "+intelligence+", "+strenght+", "+health+"]";
    }
    
    protected void gotWounded() {
    
        --health;
    }
    
    public abstract float attack();
    
    public abstract boolean defend(float attack);
}
